src/veh_type.cpp

```diff
@@ -124,34 +132,73 @@ int_id<vpart_info>::int_id( const string_id<vpart_info> &id )
 {
 }
 
-// Note on the 'symbol' flag in vehicle parts -
-// the following symbols will be translated:
-// y, u, n, b to NW, NE, SE, SW lines correspondingly
-// h, j, c to horizontal, vertical, cross correspondingly
 /**
  * Reads in a vehicle part from a JsonObject.
  */
 void vpart_info::load( JsonObject &jo )
 {
-    vpart_info next_part;
+    vpart_info def;
+
+    if( jo.has_string( "copy-from" ) ) {
+        auto const base = vehicle_part_types.find( vpart_str_id( jo.get_string( "copy-from" ) ) );
+        auto const ab = abstract_parts.find( vpart_str_id( jo.get_string( "copy-from" ) ) );
+        if( base != vehicle_part_types.end() ) {
+            def = base->second;
+        } else if( ab != abstract_parts.end() ) {
+            def = ab->second;
+        } else {
+            deferred.emplace_back( jo.str() );
+        }
+        def.id = vpart_str_id( jo.get_string( "id" ) );
+    }
 
-    next_part.id = vpart_str_id( jo.get_string( "id" ) );
+    if( jo.has_string( "abstract" ) ) {
+        def.id = vpart_str_id( jo.get_string( "abstract" ) );
+    } else {
+        def.id = vpart_str_id( jo.get_string( "id" ) );
+    }
 
-    if( jo.has_member( "name" ) ) {
-        next_part.name_ = _( jo.get_string( "name" ).c_str() );
+    assign( jo, "name", def.name_ );
```

Comment:
`vpart_info::name` doesn't translate anything. It uses the (translated) name of *item* (not vehicle part) as fallback if the vehicle part has no name itself. But that should not apply when the JSON of the vehicle part definition already contains a name.